### ISSUE TITLE
replace axios with fetch

### ISSUE THREAD

**Issue by jimmywarting** [slackapi/node-slack-sdk#1525]
(Describe your issue and goal here) 

#### Packages:

Select all that apply:

- [x] `@slack/web-api`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [ ] `@slack/oauth`
- [ ] `@slack/socket-mode`
- [ ] `@slack/types`
- [ ] I don't know

### Requirements

could you replace axios with fetch instead? 

And preferable don't depend on any FormData, Blob, XHR, fetch dependency? let users bring there own instances instead b/c they exist globally in NodeJS now and if they need they can polyfill this stuff by themself. using undici, node-fetch or anything else

**Comment by seratch** [comment]
Hi @jimmywarting, thanks for the suggestion! 

Although are not planning to work on the replacement in the short term (due to our priorities right now), I agree that we can make the API client package simpler in the interest of broader use cases in the future.

**Comment by maxcountryman** [comment]
It would be great if this were structured in such a way so that it could then work with runtimes that don't support Node, e.g. Cloudflare Workers/V8 isolates.

**Comment by jimmywarting** [comment]
often when i code things then i have this mindset of coding things in a more cross env friendlier way and i'm also following this guidelines: https://github.com/cross-js/cross-js

The most fundamentals is to think of your application as an ["onion"](https://www.codeguru.com/csharp/understanding-onion-architecture/) architecture that has different layers of supports.
at the very core of your application you shouldn't assume that you have access to any NodeJS, Deno, or any browser API's. 
it shouldn't even assume that it has system IO (network, disc) access your internet may be down (offline) perhaps you may want to cache things or mock some data or configure the request/response before it's sent & received. maybe you want to listen to [navigator.onLine](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine) and re-send the request once it's back up again. 

```js
// Just one idea:
new WebClient({
  onfetch(url, config) {
    config.signal = signal // add your own abort signal
    config.agent = agent // useful in NodeJS only
    // check if it exist in localStorage and if so return a new Response() manually
    return fetch(url, config)
  }
})
```

**Comment by filmaj** [comment]
I agree this is a great suggestion, as the use of axios makes this package less portable (to other runtimes like Deno, among other use cases). However, axios provides many different configuration options, like proxy support, which, if I understand correctly, is an important option used by many Slack application developers today (particularly in corporate environments).

So whatever replacement would be adequate, I just wanted to explicitly point out the need to maintain some amount of backwards compatibility for these secondary features like proxy support.

**Comment by alexbjorlig** [comment]
And security 🧐

Snyk just posted [this CSRF issue](https://security.snyk.io/vuln/SNYK-JS-AXIOS-6032459) with Axios.

**Comment by filmaj** [comment]
Worth revisiting this issue as I am inching towards a new major release of web-api (with things like [better type safety of API arguments](https://github.com/slackapi/node-slack-sdk/pull/1673)). In the next release, web-api will require node v18 minimum.

I wonder if in node 18+ it is easier to implement some of the more corporate-y features web-api provides like proxies.

**Comment by filmaj** [comment]
An update here is that axios 1.7 has a `fetch` adapter now 😮 so maybe we can keep axios, maintaining backwards compatibility, but also leveraging `fetch` where available? https://github.com/axios/axios/releases/tag/v1.7.0

**Comment by filmaj** [comment]
> let users bring there own instances instead b/c they exist globally in NodeJS now and if they need they can polyfill this stuff by themself. using undici, node-fetch or anything else

To trick to resolving this issue and enabling ☝️ , I suppose, is the balance between:

1. Providing some kind of plugin or extension API to let developers pick their own http client (enabling runtime agnosticism), as well as
2. Either:
    a. Maintaining backwards compatibility on certain APIs `web-api` provides like proxying et all, or
    b. Deciding on dropping those features

2.b. sounds unreasonable to me as many Slack customers heavily rely on this feature.

Given that, I am not sure what approach could be taken to both support the more corporate-y / proxy requirements as well as enabling an http client adapter / override ability for developers. Open to suggestions, though!

**Comment by eborden** [comment]
I'm curious what the current thought space is on this ticket. `axios` has a high vulnerability churn and is quickly becoming a liability within our system. How are maintainers of this library feeling about the viability of a switch?

**Comment by amotarao** [comment]
The recent supply chain compromise (axios/axios#10604) is a good reminder that `axios` is a persistent security liability. Replacing it with native `fetch` would eliminate this category of risk entirely.

**Comment by zimeg** [comment]
👋 We revisited this with recent releases that I encourage updating to when convenient - huge thanks @WilliamBergamin!:

- [@slack/web-api@8.0.0](https://github.com/slackapi/node-slack-sdk/releases/tag/%40slack%2Fweb-api%408.0.0)
- [@slack/webhook@8.0.0](https://github.com/slackapi/node-slack-sdk/releases/tag/%40slack%2Fwebhook%408.0.0)

Latest packages no longer require `axios` package and instead prefer **fetch** now! 🎉 

<img width="840" height="420" alt="Image" src="https://github.com/user-attachments/assets/c965dc3f-40a3-4c98-b05a-76b7ad5caa78" />
